src/cdk/schematics/ng-update/devkit-file-system.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {normalize} from '@angular-devkit/core';
+import {Tree, UpdateRecorder} from '@angular-devkit/schematics';
+import {relative} from 'path';
+import {FileSystem} from '../update-tool/file-system';
+
+/** File system that leverages the virtual tree from the CLI devkit. */
```

Comment:
Could you include in this description the context in which this class is intended to be used?